veps/sig-compute/tdx-sev-snp-proposal.md

```diff
@@ -0,0 +1,341 @@
+  either technologies.
+- Custom Attestation Services: Only focus on the enabling the capabilities
+  through the standard interfaces and not building a custom attestation
+  infrastructure.
+- SEV migration: No automatic upgrade path from SEV/SEV-ES to SEV-SNP, this
+  will be required to be done by explicit user configuration.
+- Confidential Devices Support.
+
+## Definition of Users
+- Cluster Admins: Responsible for enabling and managing SEV-SNP & TDX
+  capabilities in Kubernetes Clusters.
+- Developers: Deploys confidential workloads that require SEV-SNP & TDX.
+
+## User Stories
+- As a Cluster Admin, I want to enable TDX on cluster nodes so that tenants can
+  deploy Confidential Compute workloads with TDX enabled.
+- As a Cluster Admin, I need to ensure that my clusters have the correct
+  labeling that shows the correct TDX labels are showing up on my nodes.
+- As a Developer, I want to deploy VMs on a platform with TDX protection to
+  meet compliance requirements for memory integrity.
+- As a Developer, I want to deploy VMs without needing to understand the low
+  level INTEL technology.
+- As a Cluster Admin, I want to enable SEV-SNP on cluster nodes so that tenants
+  can deploy Confidential Compute workloads with SEV-SNP enabled.
+- As a Cluster Admin, I need to ensure that my clusters have the correct
+  labeling that shows the correct SEV and SEV-SNP labels are showing up on my
+  nodes.
+- As a Developer, I want to deploy VMs on a platform with SEV-SNP protection to
+  meet compliance requirements for memory integrity.
+- As a Developer, I want to deploy VMs without needing to understand the low
+  level AMD technology.
+
+## Repos
+
+- [KubeVirt](https://github.com/kubevirt/kubevirt)
+
+## Design
+
+### TDX Design
+A new TDX Feature Gate should be added first. Extend the node labeller to
+detect and label nodes capable of running TDX VMs, a corresponding node
+selector can be added to VM pods so they are scheduled correctly. TDX
+parameters need to be specified via the `launchSecurity` element of the Libvirt
+domain xml:
+```xml
+   <domain>
+     ...
+     <launchSecurity type='tdx'>
+       <policy>0x10000001</policy>
+       <mrConfigId>xxx</mrConfigId>
+       <mrOwner>xxx</mrOwner>
+       <mrOwnerConfig>xxx</mrOwnerConfig>
+       <quoteGenerationSocket path="/var/run/tdx-qgs/qgs.socket"/>
+     </launchSecurity>
+     ...
+   </domain>
+```
+All parameters are optional. Note the Libvirt support is not ready in upstream,
+refer to the
+https://github.com/intel/libvirt-tdx/commit/962325568e67014c2ff2e41a673374fafd44e385
+for the explanation of the parameters. From the implementation perspective,
+extend the VMI spec to map to the parameters:
+
+```xml
+   spec:
+     domain:
+       launchSecurity:
+         tdx:
+           policy: 0x10000000
+           mrConfigId: xxx
+           mrOwner: xxx
+           mrOwnerConfig: xxx
+           qgs: true
```

Comment:
I don't see much value for this `qgs` flag. what would be the case when this is set to `false`?

Reply:
Qemu runs OK even when the socket is not available. The socket is only needed when the guest VM issues GetQuote request Qemu needs to service.

Reply:
maybe @zhencliu can comment on this.

Reply:
+1 to @mythi that's not sumething a user has to care about. It may make sense to allow configuring in the main KubeVirt CR a QGS socket or not. In both cases you get quotes the same way inside the guest. Only difference, in one way you can get a quote inside the VM in the other case you can't get a quote. Also admins have to register the host so that the quotes are verifyable. Really not something which should be configured on the VM.

Reply:
Hi @mythi and @rmohr, just to make sure I'm following correctly: if QGS is configured in the main KubeVirt CR, would this be a global setting that enables it for all TDX VMs?

If the answer is yes, then my follow-up question: Will there be a way for a user to launch a TDX VM without QGS for use cases where it's not required(e.g. people don't care about attestation, just start a TDX VM)?"

Reply:
@zhencliu it's possible to start a TD VM with `quote-generation-socket` settings added even if the guest has no plans to use it.

Reply:
Thanks, @mythi and @rmohr. Based on the discussion, I will remove the QGS flag from the API. I agree with the plan to proceed with the basic enablement for now and hold off on adding the mr configs